### PR TITLE
Devops/ant to gradle

### LINKED ISSUE #689: Migrate Build system to gradle
**Component to be refactored**
the build system, current Ant

**Purpose of refactoring**
To allow support of additional tooling and provided a better experience for onboarding new users.
Additional, gradle better supports (e.g. less manual work in the build.xml files) multiple projects which opendcs now has (the core jar and the annotations.)

**Proposed method**
Migrate build from ant to gradle, moving some things into sub projects where appropriate.

**Additional context**
While ant *is* a currently maintained tool and has served us well, now that the lobo browser dependency is gone moving to gradle gets us better integration with some newer systems as more build tool developers focus on maven or gradle.

Additionally dependabot does not work with any ivy files so by using a standard tool can can better stay on top of critical patches for our users.

### PR DESCRIPTION
**Pull request by MikeNeilson**
## Problem Description

Fixes #689. 

Currently draft as I'm waiting to call 7.0.13 "final" or the deadline established in #714 .

## Solution

I considered not moving the source into different directories initially, however that would make the build.gradle file rather convoluted so I decided to just move things now. No code changes are required for this migration so tracing the change (a directory move) should be reasonably simple for future developers if they have the need.

- Recreate appropriate build setups in gradle
- Remove Izpack in favor of simple zip/tar
- Docs generate
- Core code, Annotations moved to individual subprojects

Todo
- [ ] Move integration tests into invididual builds (subprojects)
- [ ] Integrate sonarcloud operations
- [ ] Add automation targets for artifact and release publishing
- [ ] Update the github actions.
- [ ] Update dev docs, READMEs, and other guides

## how you tested the change

Integration tests will need to pass, as well as a test usage of the new ZIP distribution in Linux and windows.

## Where the following done:

- [ ] Tests. Check all that apply:
   - [ ] Unit tests created or modified that run during ant test.
   - [ ] Integration tests created or modified that run during integration testing
         (Formerly called regression tests.)
   - [ ] Test procedure descriptions for manual testing
- [ ] Was relevant documentation updated?
- [ ] Were relevant config element (e.g. XML data) updated as appropriate

If you aren't sure leave unchecked and we will help guide you to want needs changing where.

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+project(":install").projectDir=file("install")
+
+include ":docs"
+project(":docs").projectDir=file("docs")
+
+include ":opendcs-annotations"
+project(":opendcs-annotations").projectDir=file("libs/annotations")
+
+include ":opendcs"
+project(":opendcs").projectDir=file("libs/opendcs")
```
[the remark this answers is not in the document]
Comment (MikeNeilson) [reply settings.gradle:13]: I've seen that in other places as well. I'll wait for additional comments from others, but hearing nothing I'll change it to java before finalization.

File: libs/opendcs/build.gradle [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    id 'opendcs.deps-conventions'
+    id 'opendcs.java-conventions'
+    id 'java-library'
+}
+
+configurations {
+    all {
+        exclude group: "com.oracle.database.jdbc", module: "jmscommon"
+        exclude group: "mil.army.usace.hec", module: "loading-gui-lib"
+        exclude group: "mil.army.usace.hec", module: "dssgui"
```
[the remark this answers is not in the document]
Comment (MikeNeilson) [reply libs/opendcs/build.gradle:12]: This is on a *really* old version of each. We can't really upgrade due to the current java 8 requirement.

File: libs/opendcs/build.gradle [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    implementation(libs.hec.monolith) {
+        exclude group: "codebase"
+        exclude group: "org.jdom"
+        exclude group: "org.bouncycastle"
+        exclude group: "org.eclipse.jetty.aggregate"
+        exclude group: "javax.servlet"
+        exclude group: "mil.army.usace.hec.swingx"
+        exclude group: "com.fifesoft"
+        exclude group: "org.python"
+        exclude group: "com.oracle.database.jdbc"
```
[the remark this answers is not in the document]
Comment (MikeNeilson) [reply libs/opendcs/build.gradle:78]: I'll have to check this again, these were originally created during a game of whack-a-mole.

File: libs/opendcs/build.gradle (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamkorynta) [reply libs/opendcs/build.gradle:12]: I removed these and ran the `./gradlew :opendcs:dependencies` task and didn't see them in the tree

**Review by ktarbet — COMMENTED** [review]
It is nice to see duplicate index.rst removed.

**Comment by MikeNeilson** [comment]
> It is nice to see duplicate index.rst removed.

Yeah. I only noticed it while working on this.

**Comment by MikeNeilson** [comment]
Okay, I think I've got everything back into place, that were going to bring along anyways. I'm going to leave this as draft until we get 7.0.13 settled out a bit more but it's ready for more commentary.

Intentionally not add:
- izpack: we decided to move to a simple zip.
- automated push of artifacts: to be done in a separate PR. 
- SonarCloud integration: to be done in a separate PR.

**Pull request closed without merging**